internal/processing/processing.go

```diff
@@ -208,3 +246,81 @@ func (f *Factory) applyObjDiff(ctx context.Context, objToPatch *objToPatch) erro
 
 	return nil
 }
+
+func accessRulePatch(arPatch map[string]*objToPatch, actualState, requiredState *State) map[string]*objToPatch {
+	for path, rule := range requiredState.accessRules {
+		rulePatch := &objToPatch{}
+
+		if actualState.accessRules[path] != nil {
+			rulePatch.action = "update"
+			modifyAccessRule(actualState.accessRules[path], rule)
+			rulePatch.obj = actualState.accessRules[path]
+		} else {
+			rulePatch.action = "create"
+			rulePatch.obj = rule
+		}
+
+		arPatch[path] = rulePatch
+	}
+
+	for path, rule := range actualState.accessRules {
+		if requiredState.accessRules[path] == nil {
+			objToDelete := &objToPatch{action: "delete", obj: rule}
+			arPatch[path] = objToDelete
+		}
+	}
+
+	return arPatch
+}
+
+func authorizationPolicyPatch(apPatch map[string]*objToPatch, actualState, requiredState *State) map[string]*objToPatch {
```

Comment:
it applies all changes to the referenced `apPatch map[string]*objToPatch` returning it from the method is redundant, do we really need to return it?